mzLib/Proteomics/AminoAcidPolymer/AminoAcidPolymerExtensions.cs

```diff
@@ -73,27 +74,17 @@ public static int[] GetSequenceCoverage(this AminoAcidPolymer baseSequence, IEnu
             return bits;
         }
 
-        private static Regex _invalidResidueRegex;
-        public static Regex InvalidResidueRegex
-        {
-            get
-            {
-                if (_invalidResidueRegex == null)
-                    _invalidResidueRegex = new Regex(@"[BJXZa-z\s\p{P}\d]");
-                return _invalidResidueRegex;
-            }
-        }
-
         /// <summary>
         /// Checks whether a given string represents a valid peptide sequence without modifications. 
-        /// Valid sequences contain only uppercase letters that represent amino acids. 
+        /// Valid sequences contain only residues in the ResidueDictionary.
+        /// Note: the residue dictionary can be externally modified. Unusual amino acid letters can and do appear in peptide sequences.
         /// </summary>
         /// <param name="baseSequence"> Sequence to be checked </param>
         /// <returns> True if the sequence is valid. False otherwise. </returns>
-        public static bool ValidBaseSequence(this string baseSequence)
+        public static bool AllSequenceResiduesAreValid(this string baseSequence)
         {
             if (baseSequence.IsNullOrEmpty()) return false;
-            return !InvalidResidueRegex.IsMatch(baseSequence);
+            return !baseSequence.ToCharArray().Distinct().Except(Residue.ResiduesDictionary.Values.Select(l => l.Letter)).Any();
```

Comment:
I really like this solution. Could you split it up into a couple lines and add a comment explainer? Then I'll approve